markdown/developers/modules.md

```diff
@@ -518,6 +518,35 @@ process {
 
 5. Each output file SHOULD be emitted in it's own channel, along with the `meta` map if provided ( the exception is the versions.yml ).
 
+### Documentation
+
+1. Each module MUST have a `meta.yaml` in the same directory as the `main.nf` of the module itself.
+
+2. Keywords SHOULD be sufficient to make the module findable through research domain, data types, and tool function keywords
+
+   - Keywords MUST NOT just be solely of the (sub)tool name
+
+3. Keywords MUST be all lower case
+
+4. Input and Output sections of the meta.yaml SHOULD only have entries of input and output channels
+
+5. Tuples input and outputs MUST be split into separate entries
+
+   - i.e., `meta` should be a separate entry to the `file` it is associated with
+
+6. Input/output types MUST only be of the following categories: `string`, `file`, `directory`.
```

Comment:
What about number? https://github.com/nf-core/modules/blob/7e89020bbbade6b032c11219b723d11f4a4bbc66/modules/nf-core/gfastats/meta.yml#L32-L34

Reply:
Agreed! But should it just be generic number? or `integer`/`float` (given I've personally specified file/directory)? We could just have string/number (accounts for both integer/float)/file (both files and directories)

Reply:
Integer/float then in that case.